connector/spanmetricsconnector/internal/metrics/metrics.go

```diff
@@ -64,27 +69,50 @@ type exponentialHistogram struct {
 
 type BuildAttributesFun func() pcommon.Map
 
-func NewExponentialHistogramMetrics(maxSize int32, maxExemplarCount *int) HistogramMetrics {
+func NewExponentialHistogramMetrics(maxSize int32, maxExemplarCount *int, cardinalityLimit int) HistogramMetrics {
 	return &exponentialHistogramMetrics{
 		metrics:          make(map[Key]*exponentialHistogram),
 		maxSize:          maxSize,
 		maxExemplarCount: maxExemplarCount,
+		cardinalityLimit: cardinalityLimit,
 	}
 }
 
-func NewExplicitHistogramMetrics(bounds []float64, maxExemplarCount *int) HistogramMetrics {
+func NewExplicitHistogramMetrics(bounds []float64, maxExemplarCount *int, cardinalityLimit int) HistogramMetrics {
 	return &explicitHistogramMetrics{
 		metrics:          make(map[Key]*explicitHistogram),
 		bounds:           bounds,
 		maxExemplarCount: maxExemplarCount,
+		cardinalityLimit: cardinalityLimit,
 	}
 }
 
-func (m *explicitHistogramMetrics) GetOrCreate(key Key, attributesFun BuildAttributesFun, startTimestamp pcommon.Timestamp) Histogram {
+func (m *explicitHistogramMetrics) IsCardinalityLimitReached() bool {
+	return m.cardinalityLimit > 0 && len(m.metrics) >= m.cardinalityLimit
+}
+
+func (m *explicitHistogramMetrics) GetOrCreate(key Key, attributesFun BuildAttributesFun, startTimestamp pcommon.Timestamp) (Histogram, bool) {
+	limitReached := false
 	h, ok := m.metrics[key]
 	if !ok {
+		attributes := pcommon.NewMap()
```

Comment:
Would it make sense to just declare the variable here and create a new map only in the case where the cardinality limit is reached and the overflow attribute is not there?